File: commons/test-utils/src/main/java/com/cedarsoft/test/utils/DisableIfHeadless.java
package com.cedarsoft.test.utils;

import org.apiguardian.api.API;
import org.junit.jupiter.api.extension.*;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static org.apiguardian.api.API.Status.STABLE;

/**
 * Disables tests when run headless
 *
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
@ExtendWith(DisableIfHeadlessCondition.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@API(status = STABLE, since = "5.1")
public @interface DisableIfHeadless {
}

File: commons/test-utils/src/main/kotlin/com/cedarsoft/test/utils/DisableIfHeadless.kt
package com.cedarsoft.test.utils

import org.junit.jupiter.api.extension.ExtendWith

/**
 * Disables tests when run headless
 *
 */
@ExtendWith(DisableIfHeadlessCondition::class)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class DisableIfHeadless
